### DISCUSSION TITLE
Keeping Alpine in sync inside a CMS

### DISCUSSION THREAD

**Discussion by tslocke** [alpinejs/alpine#4118] [1. Help]
I'm developing a CMS which uses Alpine to provide some interactivity in the generated site.

There is a real-time preview. e.g. the preview updates instantly as you edit content via the CMS UI.

Some dynamic Alpine content doesn't stay in sync however. e.g. with x-for. The CMS will be updating the `<template>`, so in order to see those changes, I need to tell Alpine to regenerate the elements created by the x-for. That's one example, there could be many.

Is there a recommended public API for this situation?

Thanks

**Comment by ekwoka** [comment]
Sure!

Depending on how fine grained your system is, you can just pick a reasonable parent (the section? component? even body?) and call `Alpine.destroyTree` on it and then `Alpine.initTree`

I recommend doing `destroyTree` making whatever changes, then `Alpine.initTree`

this is what is done on https://awesomealpine.com/play to handle the in place updates.
